src/libtorch.cc

```diff
@@ -501,7 +501,7 @@ ModelInstanceState::ModelInstanceState(
     }
   }
 
-  THROW_IF_BACKEND_INSTANCE_ERROR(ValidateInputs());
+  THROW_IF_BACKEND_INSTANCE_ERROR(ValidateInputs(expected_input_cnt));
```

Comment:
That won't be correct as `expected_input_cnt` looks at the model config already. So no input results in  `expected_input_cnt == 0` and validate function won't report error correctly

Reply:
We have an additional check for when no inputs are specified: https://github.com/triton-inference-server/pytorch_backend/blob/77adac248889f21035eb57efcae5eae8d65ef5c0/src/libtorch.cc#L604-L608

Using expected_input_cnt here is symmetric with how the [onnxruntime backend](https://github.com/triton-inference-server/onnxruntime_backend/blob/main/src/onnxruntime.cc#L1134-L1141) does it.

Reply:
Removed this check since we cannot use it for pytorch backend

Reply:
In ORT it is comparing the model config to the model metadata, which is not the same in PyTorch